tooling/Microsoft.VisualStudio.BlazorExtension/AutoRebuild/BuildEventsWatcher.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.BlazorExtension
+{
+    /// <summary>
+    /// Watches for Blazor project build events, starts new builds, and tracks builds in progress.
+    /// </summary>
+    internal class BuildEventsWatcher : IVsUpdateSolutionEvents2
+    {
+        private const string BlazorProjectCapability = "Blazor";
+        private readonly IVsSolution _vsSolution;
+        private readonly IVsSolutionBuildManager _vsBuildManager;
+        private readonly object mostRecentBuildInfosLock = new object();
+        private readonly Dictionary<string, BuildInfo> mostRecentBuildInfos = new Dictionary<string, BuildInfo>();
```

Comment:
This shold probably use `OrdinalIgnoreCase`. I wouldn't expect it to happen, but if the build manager used inconsistent casings (legal on windows) that would result in some tasks that hang around forever as 'in progress' builds.

Reply:
OK will do